### PR TITLE
Feature/user login

### PR DESCRIPTION
**Pull request by lunify**
This is still a work in progress. It needs:
- [x] rebasing on top of registration (or vice versa)
- [x] commit breaking/sorting
- [x] styling
- Bonus if there is time later: frontend and integration testing (which isn't as important now with our schedule)

I added a simple schema temporarily to be able to test the code without mocking. The good news is the deployment worked with the new mongo link.
I also added authentication caching while at it as well as a centralized api service.

Hopefully by tonight RWD and the commit history will be ready. I am available to assist in registration too.

Feel free to discuss or point out anything.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -15,8 +15,10 @@
     "extends": "react-app"
   },
   "dependencies": {
+    "axios": "^0.19.0",
```
Comment (yapanss) [inline client/package.json]: I used fetch API instead in the registration component

File: client/package.json [hunk 2]
```diff
@@ -15,8 +15,10 @@
...
     "react": "^16.8.6",
     "react-dom": "^16.8.6",
+    "react-router-dom": "^5.0.1",
```
Comment (yapanss) [inline client/package.json]: This is OK

File: client/package.json (hunk 1 above)
Comment (lunify) [reply client/package.json]: Yeah, I noticed :+1:  
I can make a commit to switch to axios if you want me to do so.

File: client/src/services/api.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+// abstracted to make testing easier
+import axios from 'axios';
+
+var api, isLoggedIn;
+
+// fallback baseURL string value is because of package.json's proxy entry
+function init({
+  token = window.localStorage.getItem('token'),
+  baseURL = (api && api.defaults.baseURL) || '/api'
+} = {}) {
```
Comment (yapanss) [inline client/src/services/api.js]: It seems that the baseUrl starts with localhost:3000 instead of localhost:3001

Comment (lunify) [reply client/src/services/api.js]: Yes, the frontned starts at localhost:3000 (react's default port), and our backend uses localhost:3001 (on `index.js - line 30: const PORT = process.env.PORT || 3001`)

**Comment by lunify** [comment]
This is ready for merge.

**Pull request merged**